src/include/parser/postgresparser.h

```diff
@@ -0,0 +1,107 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// postgresparser.h
+//
+// Identification: src/include/parser/postgresparser.h
+//
+// Copyright (c) 2015-16, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "parser/statements.h"
+#include "parser/pg_query.h"
+#include "parser/parser.h"
+#include "parser/parsenodes.h"
+
+namespace peloton {
+namespace parser {
+
+//===--------------------------------------------------------------------===//
+// PostgresParser
+//
+// This is the parser that gets a Postgres parse tree first and then
+// transform into a Peloton parse tree.
+// To start working on PostgresParser:
+// First get familiar with List and ListCell structures defined in
+// pg_list.h, then take a look at the Postgres types defined in nodes.h.
+//
+// To add support for a new type of Statement:
+// Find corresponding parsenode in
+// /third_party/libpg_query/src/postgres/include/nodes/parsenodes.h or
+// /third_party/libpg_query/src/postgres/include/nodes/primnodes.h and
+// copy to src/include/parser/parsenodes.h, then add a helper function
+// for the statement.
+//
+//===--------------------------------------------------------------------===//
+
+class PostgresParser {
+ public:
+  PostgresParser();
+  ~PostgresParser();
+
+  // Parse a given query
+  static PgQueryInternalParsetreeAndError ParseSQLString(const char* sql);
+  static PgQueryInternalParsetreeAndError ParseSQLString(
+      const std::string& sql);
+
+  static PostgresParser& GetInstance();
+
+  std::unique_ptr<parser::SQLStatementList> BuildParseTree(
+      const std::string& query_string);
+
+ private:
+  // transform helper for Alias parsenodes
+  char* AliasTransform(Alias* root);
+
+  // transform helper for RangeVar parsenodes
+  parser::TableRef* RangeVarTransform(RangeVar* root);
+
+  // transform helper for JoinExpr parsenodes
+  parser::JoinDefinition* JoinTransform(JoinExpr* root);
+
+  // transform helper for from clauses
+  parser::TableRef* FromTransform(List* root);
+
+  // transform helper for select targets
+  std::vector<expression::AbstractExpression*>* TargetTransform(List* root);
+
+  // transform helper for A_Expr nodes
```

Comment:
There're some documentations in the original libpg_query code already. If I add more it's gonna be a bit redundant.